driver/api/impl/impl.cpp

```diff
@@ -1151,6 +1245,10 @@ SQLRETURN fillBinding(
         binding_info.c_type = record.getAttrAs<SQLSMALLINT>(SQL_DESC_CONCISE_TYPE, SQL_C_DEFAULT);
     }
 
+    if (binding_info.c_type == SQL_C_DEFAULT) {
+        binding_info.c_type = convertSQLTypeToCType(statement.getTypeInfo(result_set.getColumnInfo(column_idx).type_without_parameters).sql_type);
```

Comment:
Could you accidentally end up with `SQL_C_DEFAULT` as a result? Is that an error ?

Reply:
In practice, that won't happen. Column info is populated from the query execution info, and unknown types are detected there. In theory, `SQL_C_DEFAULT` will result in hard error during data extraction, so will be detected and reported in one form or another.

Reply:
Is there any test you can make to validate that it will be handled properly ?

Reply:
Not in C++ client tests.